### DISCUSSION TITLE
Improve/Clarify SynchronizationEventArgs

### DISCUSSION THREAD

**Discussion by Jan-NET** [Azure/azure-mobile-apps#589] [Ideas]
If I log the synchronization events according to the documentation (https://learn.microsoft.com/en-us/azure/developer/mobile-apps/azure-mobile-apps/howto/client/dotnet#advanced-topics), I get these logs for syncing one item based on `SynchronizationEventArgs`:

```
09:47:49:230	[0:] [SYNC] <-> todoitem, PullStarted, ItemsProcessed -1, Success False
09:47:50:819	[0:] [SYNC] <-> todoitem, ItemWillBeStored, ItemsProcessed 0, Success False
09:47:50:828	[0:] [SYNC] <-> todoitem, ItemWasStored, ItemsProcessed 1, Success False
09:47:50:858	[0:] [SYNC] <-> todoitem, PullFinished, ItemsProcessed 1, Success False
```

First I was confused about the permanent `Success = false `output. 
A look in the source code tells me `IsSuccessful` is declared as a result for a push event only. 
So, all non-push event types output confusing states of success. 
Maybe it might make sense to
1. rename this property (`IsSuccessful` --> `IsPushSuccessful`), 
2. or at least point out the specifics in the documentation 
3. or enhance `IsSuccessful `for the correct statement for the related action?

**Comment by adrianhall** [comment]
Sounds like a solid plan.  Renaming the property is a non-starter now (as it is a breaking change).  Why don't you file an issue with a proposal, and I'll get to it.
